include/revng/ADT/GenericGraph.h

```diff
@@ -10,8 +10,73 @@
 #include "llvm/ADT/STLExtras.h"
 #include "llvm/ADT/SmallVector.h"
 
+#include "revng/ADT/STLExtras.h"
 #include "revng/Support/Debug.h"
 
+/// GenericGraph is our implementation of a universal graph architecture.
+/// First and foremost, it's tailored for the revng codebase, but maybe you
+/// can find it useful as well, that's why it's released under the MIT License.
+///
+/// To use the Graph, you need to make you choice of the node type. We're
+/// currently supporting three architectures:
+///
+///   - ForwardNode - the most trivial one. It uses the least memory, but each
+///                   node only stores links with its successors. That makes
+///                   the inverse graph iteration expensive.
```

Comment:
I'd quickly summarize what the node contains.
Also, instead of saying that the inverse iteration is expensive, which is a bit misleading, I'd say that given a node it's not possible to proceed backward without a reference to the whole graph.

Reply:
What about now?